src/lib/mina_generators/parties_generators.ml

```diff
@@ -244,40 +242,44 @@ let gen_fee (account : Account.t) =
 let fee_to_amt fee =
   Currency.Amount.(Signed.of_unsigned (of_fee fee) |> Signed.negate)
 
-let gen_balance_change ?permissions_auth (account : Account.t) =
+let gen_balance_change ?permissions_auth (account : Account.t) ~new_account =
   let open Quickcheck.Let_syntax in
   let%bind sgn =
-    match permissions_auth with
-    | Some auth -> (
-        match auth with
-        | Control.Tag.None_given ->
-            return Sgn.Pos
-        | _ ->
-            Quickcheck.Generator.of_list [ Sgn.Pos; Neg ] )
-    | None ->
-        Quickcheck.Generator.of_list [ Sgn.Pos; Neg ]
+    if new_account then return Sgn.Pos
+    else
+      match permissions_auth with
+      | Some auth -> (
+          match auth with
+          | Control.Tag.None_given ->
+              return Sgn.Pos
+          | _ ->
+              Quickcheck.Generator.of_list [ Sgn.Pos; Neg ] )
+      | None ->
+          Quickcheck.Generator.of_list [ Sgn.Pos; Neg ]
   in
   (* if negative, magnitude constrained to balance in account
-         the effective balance is either what's in the account state table,
-         if provided, or what's in the ledger
+     the effective balance is what's in the account state table,
   *)
   let effective_balance = account.balance in
   let small_balance_change =
     (*make small transfers to allow generating large number of parties without an overflow*)
     let open Currency in
-    if Balance.(effective_balance < of_formatted_string "1.0") then
-      failwith "account has low balance"
+    if
+      Balance.(effective_balance < of_formatted_string "1.0") && not new_account
+    then failwith "account has low balance"
     else Balance.of_formatted_string "0.000001"
   in
   let%map (magnitude : Currency.Amount.t) =
-    Currency.Amount.gen_incl Currency.Amount.zero
-      (Currency.Balance.to_amount small_balance_change)
+    if new_account then
+      Currency.Amount.gen_incl
+        (Currency.Amount.of_formatted_string "2.0")
```

Comment:
If we reused new accounts in the same transaction then you might want to increase the initial balance here

Reply:
I gave it a small balance because the balance change in party is pretty small. I can make it bigger.